pkg/sdk/roles_integration_test.go

```diff
@@ -0,0 +1,323 @@
+package sdk
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInt_Roles(t *testing.T) {
```

Comment:
this is one long function... is it really not possible to break this up into helper functions?

e.g.
```
func TestInt_Roles(t *testing.T){
// setup work
tests = []func(t *testing.T, ...) error {Test1, Test2, Test3}
for test, _ := range tests {
t.Run(test)
}
}
```

Reply:
It sounds like this custom framework to encapsulate common test cases, which should be done in its own pr + we discussed that breaking up into multiple functions is the same as breaking up the tests with t.Run() and with one function we can create resources (like database) use it across the tests and then release these resources at the end of all tests